tests/unit/UsersControllerTest.php

```diff
@@ -1777,6 +1780,551 @@ public function testCreateSuccessfulWithValidEmailAdmin() {
+		$message->expects($this->once())
+			->method('setFrom');
+
+		$this->mailer->expects($this->once())
+			->method('createMessage')
+			->willReturn($message);
+		$this->mailer->expects($this->once())
+			->method('send')
+			->with($message);
+		$subadmin = $this->createMock(SubAdmin::class);
+		$subadmin->expects($this->once())
+			->method('getSubAdminsGroups')
+			->with($targetUser)
+			->will($this->returnValue([]));
+		$this->groupManager->method('getSubAdmin')
+			->will($this->returnValue($subadmin));
+
+		$response = $this->createController()->create('foobazz', '', [], 'validMail@Adre.ss');
+		$this->assertEquals(Http::STATUS_CREATED, $response->getStatus());
+	}
+
+	public function testSetPasswordForm() {
+		$user = $this->createMock(IUser::class);
+
+		$this->userManager->expects($this->once())
+			->method('get')
+			->with('foo')
+			->willReturn($user);
+
+		$this->config->expects($this->once())
+			->method('getUserValue')
+			->willReturn('1234:fooBaZ1');
+
+		$this->timeFactory->expects($this->once())
+			->method('getTime')
+			->willReturn(44430);
+
+		$this->urlGenerator->expects($this->once())
+			->method('linkToRouteAbsolute')
+			->willReturn('http://localhost/apps/user_management/setpassword/form/1234/foo');
+		$result = $this->createController()->setPasswordForm('fooBaZ1', 'foo');
+		$this->assertEquals(new Http\TemplateResponse(
+			'user_management', 'new_user/setpassword',
+			['link' => 'http://localhost/apps/user_management/setpassword/form/1234/foo'],
+			'guest'), $result);
+	}
+
+	public function providesUserTokenExceptionData() {
+		return [
+			[UserTokenException::TOKEN_AND_TIME_INVALID],
+			[UserTokenException::TOKEN_TIME_EXPIRED],
+			[UserTokenException::TOKEN_MISMACTH]
+		];
+	}
+
+	/**
+	 * @expectedException \OCA\UserManagement\Exception\UserTokenException
+	 * @param $tokenException
+	 * @dataProvider providesUserTokenExceptionData
+	 */
+	public function testCheckPasswordSetTokenException($tokenException) {
+		$user = $this->createMock(IUser::class);
+
+		$this->userManager->expects($this->once())
+			->method('get')
+			->with('foo')
+			->willReturn($user);
+
+		if ($tokenException === UserTokenException::TOKEN_AND_TIME_INVALID) {
+			$this->config->expects($this->once())
+				->method('getUserValue')
+				->willReturn('');
+		} elseif ($tokenException === UserTokenException::TOKEN_TIME_EXPIRED) {
+			$this->config->expects($this->once())
+				->method('getUserValue')
+				->willReturn('1234:fooBaZ1');
+			$this->timeFactory->expects($this->once())
+				->method('getTime')
+				->willReturn(44444);
+		} elseif ($tokenException === UserTokenException::TOKEN_MISMACTH) {
+			$this->config->expects($this->once())
+				->method('getUserValue')
+				->willReturn('1234:fooBaZ11');
+			$this->timeFactory->expects($this->once())
+				->method('getTime')
+				->willReturn(44430);
+		}
+		$this->invokePrivate($this->createController(), 'checkPasswordSetToken', ['fooBaZ1', 'foo']);
```

Comment:
Recheck if you really need to test this private method. Ideally you should only test public methods, and the private (and protected) methods should be fully covered by testing the public methods.
The only valid reason to do this is the cost it takes to cover all these cases from the public methods, or if this code is complex enough to require tests by itself.

Reply:
Removed. Its getting covered in other tests.